### ISSUE TITLE
EasyImgur | Kernelbase.dll crash

### ISSUE THREAD

**Issue by RitzyBusiness** [bkeiren/EasyImgur#46]
Something must of changed in my system as of recently but I am getting the following two errors when trying to run EzImgur.  I've actually re downloaded it and run it where I had extracted the file and it run fine but exhibited the same problem as soon as I authorized the app.  So the problem is when the app is linked with imgur.

Steps i've used to attempt and remedy the issue.

- I've verified .net 3.5 is still installed on my system

- I've ran sfc and dism

- I've attempted to regsrv the DLL as per a tech forum the also had no luck with that.

Error .Net Runtime Error
```
> Application: EasyImgur.exe
> Framework Version: v4.0.30319
> Description: The process was terminated due to an unhandled exception.
> Exception Info: System.ArgumentOutOfRangeException
>    at System.Threading.Thread.Sleep(System.TimeSpan)
>    at EasyImgur.ImgurAPI.TokenThread()
>    at System.Threading.ThreadHelper.ThreadStart_Context(System.Object)
>    at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
>    at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
>    at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
>    at System.Threading.ThreadHelper.ThreadStart()
```
Error Application Error
```
> Faulting application name: EasyImgur.exe, version: 0.3.2.0, time stamp: 0x557c23fb
> Faulting module name: KERNELBASE.dll, version: 10.0.15063.502, time stamp: 0xc3955624
> Exception code: 0xe0434352
> Fault offset: 0x000eb802
> Faulting process id: 0x18ec
> Faulting application start time: 0x01d317d920c27be0
> Faulting application path: C:\EasyImgur.exe
> Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
> Report Id: 777f018c-8b54-4a16-98c1-19596aff4344
> Faulting package full name: 
> Faulting package-relative application ID: 
```

**Comment by cxcorp** [comment]
Looks like the Imgur API has started issuing tokens with a much, much longer lifespan. I just tried, it issued me a token expiring in 2022! The crash is caused because earlier, EasyImgur had to refresh the access token of the Imgur API periodically, which was solved by waiting for that period and then refreshing. However, it seems that it's not possible to sleep for 1824 days 😁 

![m_TokensExpireAt = { 17/08/2022 9.46.11 }](http://i.imgur.com/L1rTJ2i.png)

@bkeiren

**Comment by bkeiren** [comment]
@cxcorp

Heh, that's funny. I'm currently away on travels but I'll try to fix this and publish a new version in about 2 weeks.

Thanks everyone :)

**Comment by RitzyBusiness** [comment]
Oh dear~ at least that means its the program/imgur is what's causing the issue. I was getting worried I broke something since I was tweaking a bunch of things in my system.

Looking forward to the update!

**Comment by cxcorp** [comment]
I could make a PR for a hotfix, just disable the token thread completely if the lifespan of the token is longer than what we can sleep?

**Comment by bkeiren** [comment]
@cxcorp

I was thinking to just enforce an upper limit to the value we pass to the Sleep() method. That way, if the Imgur API changes again we're equipped to deal with any case without risking that we never refresh the token automatically. Refreshing the tokens prematurely shouldn't be any issue because it's a very simple and cheap thing to do, so we might as well just do it every few days (or hours, weeks, whatever seems good as long as it's within the time out and the valid input to Sleep() of course).

**Comment by bkeiren** [comment]
This has been fixed in release [0.3.3](https://github.com/bkeiren/EasyImgur/releases/tag/0.3.3). Thanks everyone!
